### PR TITLE
[Merged by Bors] - feat: use local metadata for local clusters

### PR DESCRIPTION
**Pull request by galibey**
Use `LocalMetadataStorage` as a metadata backend for local clusters created by `fluvio cluster start --local`. It allows running fluvio cluster without K8.

To use K8 as a metadata backend for local clusters introduced `fluvio cluster start --local-k8` option.

### CODE REVIEW

**Review by sehz — CHANGES_REQUESTED** [review]
Overall looks good.  My main comments about make sure we don't have duplicate code and testing to ensure consistent behavior

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -748,6 +744,81 @@ jobs:
           path: diagnostics*.gz
           retention-days: 1
 
+  local_k8_cluster_test:
+    name: Local K8s cluster test run (${{ matrix.run }})-${{ matrix.test }}
+    runs-on: ${{ matrix.os }}
```
[the remark this answers is not in the document]
Comment (sehz) [reply .github/workflows/ci.yml:749]: it's not marginal run mode.  this performs full sets of partition focus test for K8

File: crates/fluvio-sc/src/start.rs [hunk 2]
```diff
@@ -57,20 +72,14 @@ fn inspect_system() {
     info!(uptime = sys.uptime(), "Uptime in secs");
 }
 
-fn inner_main_loop<C>(
```
[the remark this answers is not in the document]
Comment (sehz) [reply crates/fluvio-sc/src/start.rs:60]: remove proxy code duplication

File: .github/workflows/ci.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (galibey) [reply .github/workflows/ci.yml]: Added `local-k8` run mode to check against the full test suite.

File: crates/fluvio-sc/src/start.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (galibey) [reply crates/fluvio-sc/src/start.rs:60]: Done

**Review by sehz — APPROVED** [review]
LGTM

**Pull request closed without merging**